src/setfit/trainer.py

```diff
@@ -226,172 +206,126 @@ def call_model_init(self, params: Optional[Dict[str, Any]] = None):
 
         return model
 
-    def freeze(self):
-        """
-        Freeze SetFitModel's differentiable head.
-        Note: call this function only when using the differentiable head.
-        """
-        if not self.model.has_differentiable_head:
-            raise ValueError("Please use the differentiable head in `SetFitModel` when calling this function.")
-
-        self._freeze = True  # Currently use self._freeze as a switch
-        self.model.freeze("head")
-
-    def unfreeze(self, keep_body_frozen: bool = False):
-        """
-        Unfreeze SetFitModel's differentiable head.
-        Note: call this function only when using the differentiable head.
-
-        Args:
-            keep_body_frozen (`bool`, *optional*, defaults to `False`):
-                Whether to freeze the body when unfreeze the head.
-        """
-        if not self.model.has_differentiable_head:
-            raise ValueError("Please use the differentiable head in `SetFitModel` when calling this function.")
+    def freeze(self, component: Optional[Literal["body", "head"]] = None) -> None:
+        return self.model.freeze(component)
 
-        self._freeze = False  # Currently use self._freeze as a switch
-        self.model.unfreeze("head")
-        if keep_body_frozen:
-            self.model.freeze("body")
-        else:  # ensure to unfreeze the body
-            self.model.unfreeze("body")
+    def unfreeze(
+        self, component: Optional[Literal["body", "head"]] = None, keep_body_frozen: Optional[bool] = None
+    ) -> None:
+        return self.model.unfreeze(component, keep_body_frozen=keep_body_frozen)
 
     def train(
         self,
-        num_epochs: Optional[int] = None,
-        batch_size: Optional[int] = None,
-        learning_rate: Optional[float] = None,
-        body_learning_rate: Optional[float] = None,
-        l2_weight: Optional[float] = None,
-        max_length: Optional[int] = None,
+        args: Optional[TrainingArguments] = None,
         trial: Optional[Union["optuna.Trial", Dict[str, Any]]] = None,
-        show_progress_bar: bool = True,
+        **kwargs,
     ):
-        """
-        Main training entry point.
+        if kwargs:
+            warnings.warn(
+                f"`{self.__class__.__name__}.train` does not accept keyword arguments anymore. "
```

Comment:
Do you think we should put a deprecation warning for `v0.6` in a separate PR, e.g. this will be removed in `v1`?

Reply:
We've discussed this privately, but for others reading: we'll go with a deprecation warning starting from v1 and complete removal at v2 (or at our discretion).